core-java-modules/core-java-jpms/decoupling-pattern2/providermodule/pom.xml

```diff
@@ -17,7 +17,7 @@
     <dependencies>
         <dependency>
             <groupId>com.baeldung.servicemodule</groupId>
-            <artifactId>servicemodule2</artifactId>
+            <artifactId>servicemodule</artifactId>
```

Comment:
should this be providermodule?

Reply:
The same reason as above applies here as well i think.